src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -162,7 +160,7 @@ private void NotifySynchronizationContextOfCompletion()
             {
                 // If the interaction with the SynchronizationContext goes awry,
                 // fall back to propagating on the ThreadPool.
-                AsyncMethodBuilderCore.ThrowAsync(exc, targetContext: null);
+                Task.ThrowAsync(exc, targetContext: null);
```

Comment:
Why is "TaskType" needed, and if it's needed above, why isn't it needed here?

Reply:
(I find using a different name like this makes the code harder to understand.  It's one thing when we do something like `using Thread = Internal.Runtime.Augments.RuntimeThread;`, where `Thread` really is the known name for the thing and it's just a hack to work around another hack, but here it's introducing a new name that's not actually the right one.)

Reply:
It is to disambiguate it with Task property.

The existing place in this file used the fully qualified name `System.Threading.Tasks.Task`. I guess it would be preferable to continue doing the same.

Reply:
> It is to disambiguate it with Task property.

Ah, yes.

> The existing place in this file used the fully qualified name System.Threading.Tasks.Task. I guess it would be preferable to continue doing the same.

That would be my preference, yes.

Reply:
Yeah, that's to disambiguate. I could expand it everywhere but it's very long to type for anyone later. Before it was used at two places but now it'd be about 14 occurences.

e.g.
`
System.Threading.Tasks.Task.ThrowAsync(exception, targetContext: null);
`

Reply:
It does not look too bad to me. We have many lines that are much longer than this.

Reply:
Alright, will do another pass with that replace

Reply:
Done